File: Service/Models/Bet.cs
namespace BetSnooker.Models
{
    public class Bet
    {
        public int Id { get; set; }

        public string MatchId { get; set; }

        public int Player1Id { get; set; }

        public string Player1Name { get; set; }

        public int? Score1 { get; set; }

        public int Player2Id { get; set; }

        public string Player2Name { get; set; }

        public int? Score2 { get; set; }

        public int? WinnerId =>
            Score1.HasValue && Score2.HasValue
                ? (int?)(Score1.Value > Score2.Value ? Player1Id : Player2Id)
                : null;

        public double? ScoreValue { get; set; }

        public int? Error { get; set; }
    }
}

File: Web.UI/src/app/_models/bet.ts
export class EventBets {
    userId: string;
    roundBets: RoundBets[];
    userScore: UserScore;
}

export class UserScore {
    matchesFinished: number;
    eventScore: number;
    correctWinners: number;
    exactScores: number;
    correctWinnersAccuracy: number;
    exactScoresAccuracy: number;
    averageError: number;
    isWinner: boolean;
}

export class RoundBets {
    id: number;
    userId: string;
    updatedAt: Date;
    eventId: number;
    roundId: number;
    distance: number;
    matchBets: Bet[];

    public constructor(fields?: Partial<RoundBets>) {
      Object.assign(this, fields);
    }
}

export class Bet {
    id: number;
    matchId: number;
    active: boolean;
    player1Id: number;
    player1Name: string;
    score1?: number;
    player2Id: number;
    player2Name: string;
    score2?: number;
    scoreValue: number;
    error: number;
    betPlaced: boolean;
    matchStartDate: Date;

    formattedStartDate: string;
}
